pkg/test/conformance/stage.go

```diff
@@ -22,7 +22,9 @@ import (
 	"strconv"
 	"strings"
 
+	"gopkg.in/yaml.v2"
 	"istio.io/istio/pkg/test/conformance/constraint"
+	"istio.io/istio/pkg/test/framework/components/echo"
```

Comment:
This dependency is a bit problematic. It will end up pulling a lot of dependencies.

Reply:
Do you mean to eliminate it only from structures that model YAML files, or the test altogether? I though the assumption was different conformance test implementations would use their own Go struct definitions. Are they meant to be shared in some way (eg. copy-pasted)?

Reply:
From the YAML files. That is the data model. The data model itself should be self contained and should be purely about reading/writing these files. referencing the component code will end up pulling in all the test framework code (which I assume you don't want here as well. :) ).

Reply:
That code is already pulled in by conformance_test, so it really depends whether the code in conformance package is intended to be reused *without* conformance_test and its dependencies. I thought not, but either way doesn't hurt to not have that dependency.

Reply:
conformance_test is the actual executing test, with the particular driver implementation code which, unavoidably, ends up pulling in the dependencies. I'd expect any other alternate implementation of the APIs would end up rewriting some (or all) of this code.

The code in /pkg/... is the shared library code. That should be component agnostic.